### PR TITLE
Remove support for cabal-v1 and cabal-new

### PR DESCRIPTION
**Pull request by adithyaov**

### CODE REVIEW

File: packcheck.sh [hunk 1]
```diff
@@ -1203,22 +1163,16 @@ ensure_cabal_config() {
...
-    if test "$BUILD" = "cabal-v1"
-    then
-      echo "cabal v1-update"
-      retry_cmd cabal update
-    else
-      echo "cabal v2-update"
-      retry_cmd cabal new-update
-    fi
+    echo "cabal v2-update"
+    retry_cmd cabal new-update
```
Comment (adithyaov) [inline packcheck.sh:1175]: `v2-update` here?

File: packcheck.sh [hunk 2]
```diff
@@ -1693,10 +1627,8 @@ test -n "$1" \
     || { short_help; echo -e "\nTry --help for detailed help"; exit 1; }
 
 case $1 in
-  cabal) shift; eval_env "$@"; BUILD=cabal-v1; \
-    echo "DEPRECATED! please use cabal-v1 instead";;
-  cabal-v1) shift; eval_env "$@"; BUILD=cabal-v1; \
-    echo "DEPRECATED! please use cabal-v2 instead";;
+  cabal) die "DEPRECATED! please use cabal-v2 instead";;
+  cabal-v1) die "cabal-v1 is not supported, please use cabal-v2 instead";;
```
Comment (adithyaov) [inline packcheck.sh:1631]: `die` here because I don't want to change the behavior, nor do I want to proceed after this point.

**Review by harendra-kumar — CHANGES_REQUESTED** [review]
Remove `install_cabal_v1_deps` and wherever it is used.
Remove `cabal_v1_configure`.

**Comment by harendra-kumar** [comment]
Update the README and changelog as well.

File: README.md [hunk 3]
```diff
@@ -20,7 +20,7 @@ To use packcheck for CI testing of your repo:
 [appveyor.yml](https://github.com/composewell/packcheck/blob/master/appveyor.yml),
 or
 [.circleci/config.yml](https://github.com/composewell/packcheck/blob/master/.circleci/config.yml)
-to your package repo 
+to your package repo
```
Comment (adithyaov) [inline README.md:23]: Oops, I have a hook that removes all trailing spaces on save.

File: packcheck.sh [hunk 4]
```diff
@@ -1216,14 +1161,8 @@ ensure_cabal_config() {
...
-      if test "$BUILD" = "cabal-v1"
-      then
-        echo "cabal v1-update"
-        retry_cmd cabal update
-      else
-        echo "cabal v2-update"
-        retry_cmd cabal new-update
-      fi
+      echo "cabal v2-update"
+      retry_cmd cabal new-update
```
Comment (harendra-kumar) [inline packcheck.sh:1165]: We can rename it to `cabal v2-update`, `new-update` may go away sooner.

File: packcheck.sh [hunk 5]
```diff
@@ -1206,7 +1151,7 @@ ensure_cabal_config() {
     run_verbose cabal user-config init || cabal info . > /dev/null || true
```
Comment (harendra-kumar) [inline packcheck.sh:1151]: 1. We can now check cabal version and die if it is older than the one where we have v2 commands.
2. We can remove the `cabal info` command above.

**Review by harendra-kumar — APPROVED** [review]
You can make the suggested changes and merge.

**Comment by harendra-kumar** [comment]
We can add the github CI files from streamly and transfer the travis tests there.

**Comment by adithyaov** [comment]
> We can add the gitHub CI files from streamly and transfer the travis tests there.

Will do it in a separate PR.

**Comment by adithyaov** [comment]
@harendra-kumar Can you please review this again? I'm rather skeptical about bash scripts.

**Comment by harendra-kumar** [comment]
Looks good.

**Pull request merged**